### DISCUSSION TITLE
umi 4部署到测试环境后会出现ChunkLoadError: Loading chunk 5397 failed. 报错问题

### DISCUSSION THREAD

**Discussion by Niexx-ai** [umijs/umi#11477] [General]
![image](https://github.com/umijs/umi/assets/87113747/e1febf19-5a80-4e6b-adc5-0e5e2aa78382)

**Comment by fz6m** [comment]
需要给一个最小复现。

看看是不是 html 被配置 Nginx 缓存头了，导致发版之后还是拿到的上次的 html ，加载的旧的 js 就找不到了。

**Reply by tanabalu** [reply]
我期望在加载旧js的时候，自行重新加载一次新js，重新挂载路由。有实现途径吗

**Reply by fz6m** [reply]
不能热更新路由，必须刷新页面，可以参考 #10171 ，思路是在后台定时不断 `fetch` 没有缓存的 `index.html` ，如果发现有新版本就自动刷新页面或者提示用户新版发布刷新页面。

**Reply by Promisezhangbo** [reply]
是不是开启了mako，我开启后也会出现部分js无法加载

**Reply by hjzheng** [reply]
<img width="1685" alt="image" src="https://github.com/user-attachments/assets/a06929be-155e-4b42-ad6d-3d83b81b6f54" />
